spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresChannelMessageTableSubscription.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.channel;
+
+import org.springframework.integration.jdbc.store.JdbcChannelMessageStore;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.util.Assert;
+
+final class PostgresChannelMessageTableSubscription {
```

Comment:
I still don't see this class as a Spring Integration structure.
Your original concern was exactly about a `MessageChannel` to be ready for notification from the PostgreSQL and there is no a `MessageChannel` as a top-level Spring Integration API in your PR.
It is fully not clean why you resist to follow a simple `Subscription` contract interface and a `MessageChannel`.
Plus I said before that without an inner interface in the subscriber there is going to be a class tangle, which you exactly have just introduced with that `asSubscribableChannel()` factory method.

There is also no tests yet for this PR.
We have in the `jdbc` module a `MySqlContainerTest` to test against MySQL container in Docker.
Probably we can have something similar for Postgres to be sure in the functionality you introduce.

Reply:
I still don't understand what API you are looking for. Before that, I would not want to spend time on cleaning up the implementation or adding tests as this would require additional effort for every API suggestion. Once we find out that we're on the same page regarding the API, it's easy to add tests and clean up, that's why I set this PR as a draft.

I made the subscription class package-private to indicate that it's not the API I would go for but rather expose a factory for a SubscribableChannel. We have not discussed ChannelDispatcher before, only subscribable channel, would you rather have it as a factory for such a dispatcher?

What do you mean by "inner interface"? That the PostgresChannelMessageTableSubscriber should implement one? There's no interface in Spring integration that fits the necessary API, so I would need to create one. This goes back to what we discussed that it needs to be shared across regions and group ids as it fully occupies a physical Postgres connection.